modules/frontend/querysharding.go

```diff
@@ -154,25 +160,36 @@ func mergeResponses(ctx context.Context, rrs []RequestResponse) (*http.Response,
 
 	var errCode = http.StatusOK
 	var errBody io.ReadCloser
-	var combinedTrace []byte
-	var shardMissCount = 0
+	var combinedTrace *tempopb.Trace
+	var combinedTraceBytes []byte
+	var shardMissCount, totalBlockErrCount int
 	for _, rr := range rrs {
-		if rr.Response.StatusCode == http.StatusOK {
+		partialContent := rr.Response.StatusCode == http.StatusPartialContent
+		if rr.Response.StatusCode == http.StatusOK || partialContent {
 			body, err := io.ReadAll(rr.Response.Body)
 			rr.Response.Body.Close()
 			if err != nil {
 				return nil, errors.Wrap(err, "error reading response body at query frontend")
 			}
```

Comment:
I'm wondering if we should fail the entire request here. If we can't read the body from one of the requests, isn't this also a partial result?
Same for unmarshaling the body.